Civi/Api4/Service/Spec/Provider/OptionGroupGetSpecProvider.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+namespace Civi\Api4\Service\Spec\Provider;
+
+use Civi\Api4\Service\Spec\RequestSpec;
+use Civi\Api4\Service\Spec\FieldSpec;
+
+/**
+ * @service
+ * @internal
+ */
+class OptionGroupGetSpecProvider extends \Civi\Core\Service\AutoService implements Generic\SpecProviderInterface {
+
+  /**
+   * @param \Civi\Api4\Service\Spec\RequestSpec $spec
+   */
+  public function modifySpec(RequestSpec $spec) {
+    $field = new FieldSpec('edit_link', 'OptionGroup', 'String');
+    $field->setLabel(ts('Edit link'))
+      ->setTitle(ts('Edit link'))
+      ->setDescription(ts('Link for editing this OptionGroup'))
+      ->setColumnName('name')
+      ->setReadonly(TRUE)
+      ->setSqlRenderer([__CLASS__, 'getEditLink']);
+    $spec->addFieldSpec($field);
+
+    $field = new FieldSpec('add_link', 'OptionGroup', 'String');
+    $field->setLabel(ts('Add link'))
+      ->setTitle(ts('Add link'))
+      ->setDescription(ts('Link for editing this OptionGroup'))
+      ->setColumnName('name')
+      ->setReadonly(TRUE)
+      ->setSqlRenderer([__CLASS__, 'getAddLink']);
+    $spec->addFieldSpec($field);
+  }
+
+  /**
+   * @param string $entity
+   * @param string $action
+   *
+   * @return bool
+   */
+  public function applies($entity, $action) {
+    return $entity === 'OptionGroup' && $action === 'get';
+  }
+
+  /**
+   * Generate SQL for age field
+   * @param array $field
+   * @return string
+   *
+   *   If there is explicit handling defined in the menus, use that
+   *   otherwise, use the new SK edit
+   */
+  public static function getEditLink(array $field): string {
+    return "COALESCE(
+      (SELECT path FROM `civicrm_menu` WHERE path = CONCAT('civicrm/admin/options/', {$field['sql_name']})),
+      CONCAT('civicrm/admin/options/edit#/?gid=', `a`.`id`)
```

Comment:
@aydun this is a very clever trick but I'm not entirely sold on the idea of these calc fields. Once we add something to the API there's an expectation that it will be supported in perpetuity, but this feels more like a temporary workaround to our messy menu situation. I feel like we just need to clean up the mess.

Reply:
Ok @aydun I want to put some thought into that but have family visiting for the holidays which is not conducive to the same, but I will try soon.

Reply:
No problem - I'm stopping in a day or so until New Year so no rush from me.